### PR TITLE
[Merge] implements `engine_executePayload`

### PR DESCRIPTION
**Pull request by tbenr**
- implements `engine_executePayload`
- removes the old `consensus_newBlock`

a part of #4346 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/schema/ExecutePayloadResponse.java [hunk 1]
```diff
@@ -36,17 +41,17 @@ public boolean equals(Object o) {
...
   @Override
   public int hashCode() {
-    return Objects.hash(valid);
+    return Objects.hash(status);
   }
 
   @Override
   public String toString() {
-    return "NewBlockResponse{" + "valid=" + valid + '}';
+    return "NewBlockResponse{" + "valid=" + status.name() + '}';
```
Comment (Nashatyrev) [inline services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/schema/ExecutePayloadResponse.java]: Makes sense to fix the name?

File: ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/ExecutionPayloadUtil.java [hunk 2]
```diff
@@ -25,10 +25,10 @@
 
   public ExecutionPayloadUtil() {}
 
-  public boolean verifyExecutionStateTransition(
+  public ExecutionEngineChannel.ExecutionPayloadStatus verifyExecutionPayload(
```
Comment (Nashatyrev) [inline ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/common/util/ExecutionPayloadUtil.java]: (nit) May be import `ExecutionPayloadStatus` ?

**Review by Nashatyrev — APPROVED** [review]
LGTM
Just a couple of minor notes

File: services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/schema/ExecutePayloadResponse.java (hunk 1 above)
Comment (tbenr) [reply services/powchain/src/main/java/tech/pegasys/teku/services/powchain/execution/client/schema/ExecutePayloadResponse.java]: Now I refreshed the how enum works, you're right :)

**Pull request merged**